### PR TITLE
Ddp 6507 saved filters fix

### PR DESCRIPTION
**Pull request by Irakli-Khaladze**
Fix newly found issue with defaultColumns, make saved filters work on participant data fields as well

### CODE REVIEW

**Review by pegahtah — APPROVED** [review]

File: src/app/filter-column/filter-column.model.ts [hunk 1]
```diff
@@ -464,6 +469,40 @@ export class Filter {
...
+    filter.parentName = jsonFilter.participantColumn.object;
+    let selectedOptions = [];
+    if (filter.selectedOptions != null && jsonFilter.options != undefined) {
+      for (let value of Object.values( jsonFilter.options )) {
+        if (value['value'] && filter.selectedOptions.includes(value['value'])) {
+          selectedOptions.push(value['value']);
+        }
+      }
+    }
+    if (filter.filter1 == undefined || filter.filter1 == null) {
```
Comment (gmakharat) [inline src/app/filter-column/filter-column.model.ts]: expression could be written like `if (!filter.filter1)`

File: src/app/filter-column/filter-column.model.ts [hunk 2]
```diff
@@ -464,6 +469,40 @@ export class Filter {
...
+      filter.filter1 = new NameValue(filter.participantColumn.name, null);
+    }
+    filter.filter1.value = this.replace(filter.filter1.value);
+    let newFilter = new Filter(filter.participantColumn, filter.type, jsonFilter.options, filter.filter2, filter.range, filter.exactMatch, filter.filter1,
+      selectedOptions, (filter.filter1 == null || filter.filter1 == undefined) ? null : filter.filter1.value,
```
Comment (gmakharat) [inline src/app/filter-column/filter-column.model.ts]: you have already checked in above if statement that if `filter.filter1` is null or undefined you set value on it, I think that at this point `filter.filter1`  will always have value.

File: src/app/filter-column/filter-column.model.ts [hunk 3]
```diff
@@ -464,6 +469,40 @@ export class Filter {
...
+      (filter.filter2 == null || filter.filter2 == undefined) ? null : filter.filter2.value, null,
```
Comment (gmakharat) [inline src/app/filter-column/filter-column.model.ts]: `!filter.filter2`

File: src/app/participant-list/participant-list.component.ts [hunk 4]
```diff
@@ -737,6 +738,9 @@ export class ParticipantListComponent implements OnInit {
       }
     );
   }
+  selectedColumnsNotEmpty(): boolean {
```
Comment (gmakharat) [inline src/app/participant-list/participant-list.component.ts]: I personally always prefer to follow convention about method which returns boolean value and use `is` prefix to give method quick understanding what it checks, or `has` if `is` does not fit.

File: src/app/participant-list/participant-list.component.ts [hunk 5]
```diff
@@ -737,6 +738,9 @@ export class ParticipantListComponent implements OnInit {
...
+    return Object.values(this.selectedColumns).find(value => value != null && value != []) !== null;
```
Comment (gmakharat) [inline src/app/participant-list/participant-list.component.ts]: what do you check here `value != []`?

Comment (Irakli-Khaladze) [reply src/app/participant-list/participant-list.component.ts]: so that it's not empty array, couldn't check with length because it didn't have type

Comment (gmakharat) [reply src/app/participant-list/participant-list.component.ts]: are you sure that such checking will work? value should be array here? if so `(value: []) => value != null && value.length > 0`

File: src/app/services/dsm.service.ts [hunk 6]
```diff
@@ -132,14 +132,14 @@ export class DSMService {
       if (json != null && json.filters != null) {
         viewFilterCopy = json.copy();
         for (let filter of json.filters) {
-          if (filter.type === Filter.OPTION_TYPE) {
+          if (filter.type === Filter.OPTION_TYPE && filter.participantColumn.tableAlias !== 'participantData') {
```
Comment (SimoneMaiwald) [inline src/app/services/dsm.service.ts:135]: did you test that change with other studies than just RGP and AT? What happens if type is option but not participantData?

**Review by SimoneMaiwald — APPROVED** [review]

**Pull request merged**